app/components/candidate_interface/offer_review_component.rb

```diff
@@ -11,6 +11,8 @@ def rows
         location_row,
       ]
       rows << conditions_row if @course_choice.offer.conditions.any?
+      rows.insert(2, fee_row) if @course_choice.current_course.fee?
+      rows.insert(2, salary_row) if @course_choice.current_course.salary? || @course_choice.current_course.apprenticeship?
```

Comment:
Never knew about `Array.insert`!

Would it be slightly more intuitive to move these lines up above the line 13 and just used the plain `<<` method?

Reply:
Not sure I follow? Don't we want the fee/salary row to be in between the course and location rows?

Reply:
@avinhurry yep, but can't you do that by rearrange the line in the source code so the array gets appended to in a different order, rather than using `Array.insert`? Or am I missing something? It's only a suggestion though, as I wasn't familiar with `Array.insert`...

Reply:
Above line 13 would be in the array? We can insert it that way and it would be in the right place, but I don't think we can use the `if` logic then, unless i'm missing something 🤔 or just completely misunderstanding 😆